ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/validate/ConflictCheckWithCommittedImpl.scala

```diff
@@ -92,25 +91,27 @@ private[validate] class ConflictCheckWithCommittedImpl(
       Future.successful(Right(()))
     else
       indexService
-        .lookupMaximumLedgerTime(referredContracts)(transaction.loggingContext)
+        .lookupMaximumLedgerTimeAfterInterpretation(referredContracts)(transaction.loggingContext)
         .transform {
-          case Failure(MissingContracts(missingContractIds)) =>
+          case Success(MaximumLedgerTime.Archived(missingContractIds)) =>
             Success(Left(UnknownContracts(missingContractIds)(completionInfo, errorFactories)))
+
           case Failure(err) =>
             Success(Left(LedgerBridgeInternalError(err, completionInfo)))
-          case Success(maximumLedgerEffectiveTime) =>
-            maximumLedgerEffectiveTime
-              .filter(_ > transactionLedgerEffectiveTime)
-              .fold[Try[Validation[Unit]]](Success(Right(())))(contractLedgerEffectiveTime =>
-                Success(
-                  Left(
-                    CausalMonotonicityViolation(
-                      contractLedgerEffectiveTime = contractLedgerEffectiveTime,
-                      transactionLedgerEffectiveTime = transactionLedgerEffectiveTime,
-                    )(completionInfo, errorFactories)
-                  )
-                )
+
+          case Success(MaximumLedgerTime.Max(maximumLedgerEffectiveTime))
+              if maximumLedgerEffectiveTime > transactionLedgerEffectiveTime =>
+            Success(
+              Left(
+                CausalMonotonicityViolation(
+                  contractLedgerEffectiveTime = maximumLedgerEffectiveTime,
+                  transactionLedgerEffectiveTime = transactionLedgerEffectiveTime,
+                )(completionInfo, errorFactories)
               )
+            )
+
+          case Success(_) =>
```

Comment:
```suggestion
          case Success(MaximumLedgerTime.NotAvailable) =>
```

to have the compiler complain if you add new cases

Reply:
That wont work: the one with the max above is with a guard. I would prefer this way, so that top cases declare errors, and there is a catch case for success.